frontend/src/metabase/visualizations/lib/renderer_utils.js

```diff
@@ -289,6 +295,18 @@ export function syntheticStackedBarsForWaterfallChart(datas, settings = {}) {
   return stackedBarsDatas;
 }
 
+export function xValueForWaterfallTotal({ settings, series }) {
```

Comment:
I think this will mirror `getXInterval` with non-timeseries quantitative scales added. Could it use the interval returned from that rather than recomputing it? 

Maybe you could thread x axis props into `getDimensionsAndGroupsAndUpdateSeriesDisplayNames`?

Reply:
Absolutely we can DRY this and share the calculation once we can move `getXInterval` to `renderer_utils.js`. I feel that it's a separate refactoring (not relevant to this PR) but I'm also happy to include it here.

Reply:
👍 makes sense to do that later.